src/test/java/org/opensearch/security/configuration/ConfigurationRepositoryTest.java

```diff
@@ -0,0 +1,134 @@
+
+        assertThat(configRepository, is(notNullValue()));
+        assertThat(configRepository, instanceOf(ConfigurationRepository.class));
+    }
+
+    @Test
+    public void initOnNodeStart_withSecurityIndexCreationEnabledShouldSetInstallDefaultConfigTrue() {
+        Settings settings = Settings.builder().put(ConfigConstants.SECURITY_ALLOW_DEFAULT_INIT_SECURITYINDEX, true).build();
+
+        ConfigurationRepository configRepository = createConfigurationRepository(settings);
+
+        configRepository.initOnNodeStart();
+
+        assertThat(configRepository.getInstallDefaultConfig().get(), is(true));
+    }
+
+    @Test
+    public void initOnNodeStart_withSecurityIndexNotCreatedShouldNotSetInstallDefaultConfig() {
+        Settings settings = Settings.builder().put(ConfigConstants.SECURITY_BACKGROUND_INIT_IF_SECURITYINDEX_NOT_EXIST, false).build();
+
+        ConfigurationRepository configRepository = createConfigurationRepository(settings);
+
+        configRepository.initOnNodeStart();
+
+        assertThat(configRepository.getInstallDefaultConfig().get(), is(false));
+    }
+
+    @Test
+    public void getConfiguration_withInvalidConfigurationShouldReturnSecurityDynamicConfigurationEmpty() throws IOException {
+        ConfigurationRepository configRepository = createConfigurationRepository(Settings.EMPTY);
+
+        SecurityDynamicConfiguration<?> config = configRepository.getConfiguration(CType.CONFIG);
+
+        assertThat(config, instanceOf(SecurityDynamicConfiguration.class));
+        assertThat(config.getCEntries().size(), is(equalTo(0)));
+    }
+
+    @Test
+    public void getConfiguration_withValidConfigurationShouldReturnDeepClone() throws IOException {
+
+        ConfigurationRepository configRepository = mock(ConfigurationRepository.class);
+
+        var objectMapper = DefaultObjectMapper.objectMapper;
+        var objectNode = objectMapper.createObjectNode();
+        objectNode.set("_meta", objectMapper.createObjectNode().put("type", CType.ROLES.toLCString()).put("config_version", 2));
+        objectNode.set("kibana_read_only", objectMapper.createObjectNode().put("reserved", true));
+        objectNode.set("some_hidden_role", objectMapper.createObjectNode().put("hidden", true));
+        objectNode.set("all_access", objectMapper.createObjectNode().put("static", true)); // it reserved as well
+        objectNode.set("security_rest_api_access", objectMapper.createObjectNode().put("reserved", true));
+
+        when(configRepository.getConfiguration(CType.ROLES)).thenReturn(
+            SecurityDynamicConfiguration.fromJson(objectMapper.writeValueAsString(objectNode), CType.ROLES, 2, 1, 1)
+        );
+
+        SecurityDynamicConfiguration<?> config2 = configRepository.getConfiguration(CType.ROLES);
+
+        assertThat(config2, is(notNullValue()));
+        assertThat(config2, instanceOf(SecurityDynamicConfiguration.class));
+        assertThat(config2.getCEntries().size(), is(greaterThan(0)));
```

Comment:
Can you add a test here to ensure that it is actually a clone and not the same copy?

Reply:
@DarshitChanpura thanks to you comment I realized that this test is not working as I expected and I spent a lot of time trying to make it work but I couldn't. This line `SecurityDynamicConfiguration<?> conf = configCache.getIfPresent(configurationType);`  always returns null since `configCache` is empty.
I tried mocking up configCache and using `when(configCache.getIfPresent...` but it didn't work either.
I would appreciate any help here.